.ci/build-n-run.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+uname -r
+#build
+cd examples
+make
+cd ../
+#run module test
+exclude_list=$(cat .ci/non-working.txt);   #the module in non-working.txt will be ignored by the shell
+for filename in examples/*.ko; do
+    [ -e "$filename" ] || continue
+    exclude=false
+    for file in $exclude_list; do
+        if [ "$filename" = "$file" ]; then
+            exclude=true
+        fi
+    done
+    
+    if [ "$exclude" = true ]; then
+        echo "$filename" "is excluded!"; continue;
+    else
+        echo "$filename"
+#test 2 times
+        ( sudo insmod "$filename" && sudo rmmod "$filename" ) || exit 1;
```

Comment:
You should trim `$filename` to remove the `.ko` suffix for `rmmod`.

Reply:
You can make use of bash function to modularize the routines.
E.g., https://github.com/sysprog21/kvm-host/blob/master/scripts/build-linux-image.sh